packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+    variant('ham', default=True, description='Build with hammoz and atm_phy_echam enabled.')
+
+    conflicts('+openmp', when='%intel')
+    conflicts('+openmp', when='%cce')
+    conflicts('+claw', when='%intel')
+    conflicts('+claw', when='%cce')
+    conflicts('icon_target=gpu', when='%intel')
+    conflicts('icon_target=gpu', when='%cce')
+
+    atm_phy_echam_submodels_namelists_dir = 'externals/atm_phy_echam_submodels/namelists'
+
+    def setup_build_environment(self, env):
+        self.setup_run_environment(env)
+        if self.spec.variants['icon_target'].value == 'gpu':
+            env.set('CUDA_HOME', self.spec['cuda'].prefix)
+
+    @run_before('configure')
+    def generate_hammoz_nml(self):
+        if '@ham' in self.spec:
+            with working_dir(self.atm_phy_echam_submodels_namelists_dir):
+                make()
+
+    def configure_args(self):
+        args = []
+
+        CFLAGS=''
+        CPPFLAGS=''
+        FCFLAGS=''
+        LDFLAGS=''
+        LIBS=''
+
+        # Icon-hammoz:
+        if '+ham' in self.spec:
+            args.append('--enable-atm-phy-echam-submodels')
+            args.append('--enable-hammoz')
+
+        # Eccodes library:
+        if '+eccodes' in self.spec:
+            args.append('--enable-grib2')
+            args.append('--without-external-cdi')
+            ECCODESI='-I' + self.spec['eccodes'].prefix + '/include '
+            ECCODESL='-L' + self.spec['eccodes'].prefix + '/lib '
+            ECCODES_LIBS='-leccodes '
+            CPPFLAGS+=ECCODESI
+            LDFLAGS+=ECCODESL
+            LIBS+=ECCODES_LIBS
+
+        # Claw library
+        if '+claw' in self.spec:
+            args.append('CLAW=' + self.spec['claw'].prefix + '/bin/clawfc')
+            args.append('CLAWFLAGS=-I' + self.spec['netcdf-c'].prefix + '/' + self.compiler.name + '/' + str(self.compiler.version.up_to(2))  + '/include')
+            args.append('--enable-claw')
+
+        # Rte-rttmgp
+        if '+rte-rrtmgp' in self.spec:
+            args.append('--enable-rte-rrtmgp')
+
+        # Mpi-checks
+        if '~mpi-checks' in self.spec:
+            args.append('--disable-mpi-checks')
+
+        # OpenMP library
+        if '~openmp' in self.spec:
+            args.append('--disable-openmp')
+        else:
+            LIBS+='-lomptarget '
+            LDFLAGS+=' -L' + self.spec['ompt-openmp'].prefix + ' '
+
+        # Serialbox library
+        if '+serialize' in self.spec:
+            SERIALBOXI='-I' + self.spec['serialbox'].prefix + '/include '
+            SERIALBOXL='-L' + self.spec['serialbox'].prefix + '/lib '
+            SERIALBOX_LIBS='-lSerialboxFortran '
+            FCFLAGS+=SERIALBOXI
+            LDFLAGS+=SERIALBOXL
+            LIBS+=SERIALBOX_LIBS
+
+            args.append('SB2PP=python2 ' + self.spec['serialbox'].prefix +  '/python/pp_ser/pp_ser.py ')
+            if self.spec.variants['icon_target'].value == 'gpu':
```

Comment:
I kind of like the fact that you do not have to worry about setting `read` or `write` mode for serialbox. However, it takes away a bit of flexibility. In principle, serialbox could also be used to compare CPU code against (different/optimized) CPU code. I would decouple gpu/cpu targets from serialbox read/write.